frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -164,6 +182,24 @@ const StrategyFormBody = ({
     }
   }, [selectedStrategyType, values, setFieldValue]);
 
+  useEffect(
+    function removeDefaultValuesFromForm() {
+      if (isSubmitting) {
+        for (const fieldName in values) {
+          const value = values[fieldName as keyof CacheStrategy];
+          const defaultValue = getDefaultValueForField(values.type, fieldName);
+          // Formik will initially record numbers as strings so we need a loose comparison
+          // eslint-disable-next-line eqeqeq
+          if (defaultValue && value == defaultValue) {
+            setFieldValue(fieldName, undefined);
+          }
+        }
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [isSubmitting],
+  );
```

Comment:
The ugliness of this `==` comparison is another reason to reconsider how default values are handled in this form.